### PR TITLE
implemented the music rental scenario

### PR DESCRIPTION
**Pull request by YFisch**
I chose the music rental scenario and did the work please review it.

### CODE REVIEW

File: 1 table sketch.sql [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    Category varchar(100) not null, not blank
+    InstrumentID computed column concat first 3 letters of the instrument and of the category
+    MonthlyRentalFee decimal (6,2), not null, must be greater than 0
+    TotalProfit computed column MonthlyRentalFee * months rented
+    DateRented Date not null, not future date
+    DateReturned not null not future date
+    DateInserted datetime not null, default getdate()
+    
+    multy constraint columns
+        DateReturned mus be after data rented
```
Comment (AshiraFriedman) [inline 1 table sketch.sql:18]: Please implement the above constraint in your table file

File: 1 table sketch.sql [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    InstrumentID computed column concat first 3 letters of the instrument and of the category
+    MonthlyRentalFee decimal (6,2), not null, must be greater than 0
+    TotalProfit computed column MonthlyRentalFee * months rented
+    DateRented Date not null, not future date
+    DateReturned not null not future date
+    DateInserted datetime not null, default getdate()
+    
+    multy constraint columns
+        DateReturned mus be after data rented 
+        either DateReturned are both blank or both full
```
Comment (AshiraFriedman) [inline 1 table sketch.sql:19]: What did you mean by DateReturned both blank or both full?

File: 1 table sketch.sql [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        between DateReturned and DateRented there can't be more than 12 months and less than 1 month
```
Comment (AshiraFriedman) [inline 1 table sketch.sql:20]: Please implement the above constraint

File: 1 table.sql [hunk 4]
```diff
@@ -0,0 +1,18 @@
+
+use musicrentalDB
+go
+drop table if exists dbo.MusicRental
+go
+create table dbo.MusicRental(
+    MusicRentalID int not null identity primary key,
+    CustomerFirstName varchar (100) not null constraint ck_MusicRental_customer_first_name_cannot_be_blank check(CustomerFirstName <> ''),   
+    CustomerLastName varchar (100) not null constraint ck_MusicRental_customer_last_name_cannot_be_blank check(CustomerLastName <> ''),
+    Instrument varchar(100) not null constraint ck_MusicRental_insrument_cannot_be_blank check (Instrument <> ''),
```
Comment (AshiraFriedman) [inline 1 table.sql:10]: Just missing the 't' in 'instrument' in the above constraint name

File: 3 reports.sql [hunk 5]
```diff
@@ -0,0 +1,20 @@
+--1) I need to know if I have any repeat customers and what they rented.
```
Comment (AshiraFriedman) [inline 3 reports.sql:1]: Include what the customer rented in the select statement

File: 3 reports.sql [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+ from MusicRental m
+ group by concat(m.CustomerFirstName, ' ', m.CustomerLastName)
+ having count(concat(m.CustomerFirstName, ' ', m.CustomerLastName)) > 1
+ order by count(concat(m.CustomerFirstName, ' ', m.CustomerLastName))
+--2) I need to know the most popular category of instrument that has ever been rented.
+select top 1 m.Category
+from MusicRental m
+group by m.Category
+order by COUNT(m.Category) desc
+--3) I need to know which instrument (that is not being currently rented) made the most profit.
```
Comment (AshiraFriedman) [inline 3 reports.sql:12]: The report just asks to return the profit and instrument, the report doesn't ask for the other data that you are returning

**Review by AshiraFriedman — COMMENTED** [review]
Wow!  Great job implementing this scenario! 
Please see comments in review, and respond when you have made the requested changes.

File: 1 table sketch.sql (hunk 1 above)
Comment (YFisch) [reply 1 table sketch.sql:18]: sorry forgot to implement my multi constraints column

File: 1 table sketch.sql (hunk 2 above)
Comment (YFisch) [reply 1 table sketch.sql:19]: I fixed this constraint it missed a few words

File: 1 table sketch.sql (hunk 3 above)
Comment (YFisch) [reply 1 table sketch.sql:20]: I implemented the constraint

File: 1 table.sql (hunk 4 above)
Comment (YFisch) [reply 1 table.sql:10]: sorry silly mistake

File: 3 reports.sql (hunk 6 above)
Comment (YFisch) [reply 3 reports.sql:12]: Fixed mistake

File: 3 reports.sql (hunk 5 above)
Comment (YFisch) [reply 3 reports.sql:1]: I can't figure out how to include what the customer rented as it is not included in the group by

**Review by YFisch — COMMENTED** [review]
I fixed all the issues, see my latest commits

File: 1 table sketch.sql (hunk 1 above)
Comment (AshiraFriedman) [reply 1 table sketch.sql]: no problem, good job

File: 1 table sketch.sql (hunk 2 above)
Comment (AshiraFriedman) [reply 1 table sketch.sql]: there is no need for this constraint, as Profit is a computed column, and you already have constraints on the columns that it is referencing, in order to ensure proper data

File: 3 reports.sql (hunk 5 above)
Comment (AshiraFriedman) [reply 3 reports.sql:3]: You are right, this question is not in the scope of your knowledge at this point in the course, you can leave it as is.  In a few sessions, you'll learn about something called a cte, which is what you would use in this case.

**Comment by YFisch** [comment]
I erased the extra constraint

**Comment by AshiraFriedman** [comment]
Impressive job!
Mark 100%

**Pull request closed without merging**